server/index.js

```diff
@@ -91,6 +91,23 @@ app.post('/api/waitlist-reservation', (req, res, next) => {
     .catch(err => next(err));
 });
 
+app.put('/api/edit-reservation', (req, res, next) => {
+  const { name, partyBefore, partyUpdate, uniqueCode } = req.body;
+  const sql = `
+  update "reservations"
+  set "partySize" = $1
+  from "restaurants"
+  where "customerName" = $2 AND "uniqueCode" = $3 AND "partySize" = $4
```

Comment:
It doesn't really matter if they have the right name or know their previous part size if they are using the correct unique code. Try to simplify the feature a bit.